src/sync/treiber_stack.rs

```diff
@@ -1,82 +1,107 @@
+
 use std::sync::atomic::Ordering::{Acquire, Release, Relaxed};
 use std::ptr;
 
-use mem::epoch::{self, Atomic, Owned};
+use epoch::{self, Atomic};
 
 /// Treiber's lock-free stack.
 ///
-/// Usable with any number of producers and consumers.
+/// This is usable with any number of producers and consumers.
 #[derive(Debug)]
 pub struct TreiberStack<T> {
+    /// The head (top) node.
     head: Atomic<Node<T>>,
 }
 
+/// A node in the stack.
 #[derive(Debug)]
 struct Node<T> {
+    /// The payload of this node.
     data: T,
+    /// The node below this node.
     next: Atomic<Node<T>>,
 }
 
 impl<T> TreiberStack<T> {
     /// Create a new, empty stack.
     pub fn new() -> TreiberStack<T> {
-        TreiberStack { head: Atomic::null() }
+        TreiberStack {
+            head: Atomic::null(),
+        }
     }
 
-    /// Push `t` on top of the stack.
-    pub fn push(&self, t: T) {
-        let mut n = Owned::new(Node {
-            data: t,
+    /// Push an element on top of the stack.
+    pub fn push(&self, elem: T) {
+        // Construct the node.
+        let mut n = Box::new(Node {
+            data: elem,
             next: Atomic::null(),
         });
+
+        // Pin the epoch.
         let guard = epoch::pin();
+        // In order to solve the ABA condition, we spin until the CAS succeeds.
         loop {
+            // Load the head node.
             let head = self.head.load(Relaxed, &guard);
+            // Store the constructed head as the tail of the constructed node.
             n.next.store_shared(head, Relaxed);
-            match self.head.cas_and_ref(head, n, Release, &guard) {
+            // CAS the head to the constructed head.
+            match self.head.compare_and_set_ref(head, n, Release, &guard) {
+                // Everything went well.
                 Ok(_) => break,
+                // It failed. Another thread changed the contents (pushed or popped) of the stack,
+                // so we must repeat the proess.
                 Err(owned) => n = owned,
             }
         }
     }
 
     /// Attempt to pop the top element of the stack.
-    /// **Deprecated method**, use try_pop
     ///
-    /// Returns `None` if the stack is observed to be empty.
-    #[cfg_attr(any(feature="beta", feature="nightly"), deprecated(note="The pop method has been renamed to try_pop for consistency with other collections."))]
+    /// This returns `None` if the stack is observed to be empty.
     pub fn pop(&self) -> Option<T> {
-        self.try_pop()
-    }
-
-    /// Attempt to pop the top element of the stack.
-    ///
-    /// Returns `None` if the stack is observed to be empty.
-    pub fn try_pop(&self) -> Option<T> {
+        // Pin the epoch.
         let guard = epoch::pin();
+        // Spin until the ABA condition is resolved.
         loop {
+            // Load the node.
             match self.head.load(Acquire, &guard) {
                 Some(head) => {
+                    // Load the tail node.
```

Comment:
Referring this to as the tail node is confusing when in other contexts such as the queue the tail is not the next pointer.